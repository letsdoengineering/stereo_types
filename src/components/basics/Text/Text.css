.text_styles-color-Gray {
    color: var(--dark-gray)
}

.text_styles-color-Blue {
    color: var(--default-blue)
}

.text_styles-size-XS {
    font-size: var(--font-size-XS);
}

.text_styles-size-S {
    font-size: var(--font-size-S);
}

.text_styles-size-M {
    font-size: var(--font-size-M);
}

.text_styles-size-L {
    font-size: var(--font-size-L);
}

.text_styles-size-XL {
    font-size: var(--font-size-XL);
}

.text_styles-weight-Normal {
    font-weight: var(--font-weight-normal);
}

.text_styles-weight-Bold {
    font-weight: var(--font-weight-bold);
}

.text_styles-weight-Xtra {
    font-weight: var(--font-weight-xtra);
}
