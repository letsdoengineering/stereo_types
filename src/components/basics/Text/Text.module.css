.text-gray {
    color: var(--dark-gray)
}

.text-blue {
    color: var(--default-blue)
}

.size-XS {
    font-size: var(--font-size-XS);
}

.size-S {
    font-size: var(--font-size-S);
}

.size-M {
    font-size: var(--font-size-M);
}

.size-L {
    font-size: var(--font-size-L);
}

.size-XL {
    font-size: var(--font-size-XL);
}

.weight-normal {
    font-weight: var(--font-weight-normal);
}

.weight-bold {
    font-weight: var(--font-weight-bold);
}

.weight-xtra {
    font-weight: var(--font-weight-xtra);
}
