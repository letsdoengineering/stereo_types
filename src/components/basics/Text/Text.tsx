import React from 'react'
import * as textStyles from './Text.module.css'
import getClassNames from '../../../utils/get-class-names'

// TODO: find appropriate interface to extend for span, p and text
type TextProps = {
  /** the type of Text element tag to render */
  tagName?: 'span' | 'p' | 'text'
  /** optional: class-names */
  className?: string
  /** optional: specify the role for accessibility purposes when semantic role has changed */
  role?: string
  /** children always of type ReactNode */
  children: React.ReactNode
  /** colour choice*/
  color?: 'Gray' | 'Blue'
  /** 16 | 18 | 20 | 30 px*/
  size?: ElementSize
  /** font weight */
  weight?: 'Normal' | 'Bold' | 'Xtra'
  id?: string
}

/** Text: renders a Text tag of a type based on the tagName prop */
const Text: React.FC<TextProps> = ({
  tagName: Wrapper = 'span',
  className,
  children,
  color = 'Gray',
  size = 'S',
  weight = 'Normal',
  id,
  ...rest
}: TextProps) => {
  const classNames = getClassNames({
    defaultClasses: [
      textStyles[`size${size}`],
      textStyles[`weight${weight}`],
      textStyles[`text${color}`],
    ],
    className,
  })

  return (
    <Wrapper className={classNames} id={id} {...rest}>
      {children}
    </Wrapper>
  )
}

export default Text
