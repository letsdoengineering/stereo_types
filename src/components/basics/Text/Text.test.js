import * as React from 'react'
import { render, screen, waitFor } from '@testing-library/react'
import Text from './Text'
import renderer from 'react-test-renderer'
import { axe } from 'jest-axe'

describe('Text', () => {
  describe('dynamic tags', () => {
    it('renders text in a span tag by default', async () => {
      render(<Text>Some text</Text>)
      const text = screen.getByText('Some text')

      await waitFor(() => expect(text.nodeName).toBe('SPAN'))
    })

    it("renders text in a p tag if tagName='p'", async () => {
      render(<Text tagName='p'>Some text</Text>)
      const text = screen.getByText('Some text')

      await waitFor(() => expect(text.nodeName).toBe('P'))
    })
  })

  it('has not changed the JSX layout of the page from previous snapshot', () => {
    const tree = renderer.create(<Text />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders the page DOM that is accessible', async () => {
    const { container } = render(<Text tagName='p'>Some text</Text>)
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
