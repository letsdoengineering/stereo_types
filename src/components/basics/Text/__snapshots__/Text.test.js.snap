// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text has not changed the JSX layout of the page from previous snapshot 1`] = `
<span
  className="sizeS weightNormal textGray"
/>
`;
